File: 0735-asteroid-collision/0735-asteroid-collision.py
class Solution:
    def asteroidCollision(self, asteroids: List[int]) -> List[int]:
        st = []
        for i in range(len(asteroids)):
            if(len(st)==0 or asteroids[i]>=0):
                st.append(asteroids[i])
            else:
                if(st[-1]>=0):
                    while(not(len(st)==0) and st[-1]<abs(asteroids[i])):
                        if(st[-1]<0):
                            break
                        st.pop()
                    if(not(len(st)==0) and st[-1]==abs(asteroids[i])):
                        st.pop()
                        continue
                    if(len(st)==0 or st[-1]<0):
                        st.append(asteroids[i])
                else:
                    st.append(asteroids[i])
        return st 

File: 0735-asteroid-collision/0735-asteroid-collision.java
class Solution {
    public int[] asteroidCollision(int[] asteroids) {
        Stack<Integer> st = new Stack<>();
        for(int i=0; i<asteroids.length; i++){
            if(st.isEmpty() || asteroids[i]>=0){
                st.push(asteroids[i]);
                // System.out.println(st);
            }
            else{
                if(st.peek()>=0){
                    while(!(st.isEmpty()) && st.peek()<Math.abs(asteroids[i])){
                        if(st.peek()<0){
                            break;
                        }
                        st.pop();
                    }
                    if(!(st.isEmpty()) && st.peek()==Math.abs(asteroids[i])){
                        st.pop();
                        continue;
                    }
                    if(st.isEmpty() || st.peek()<0){
                        st.push(asteroids[i]);
                    }
                }
                else{
                    st.push(asteroids[i]);
                }
            }
        }
        
        int[] ans = new int[st.size()];
        for(int i=ans.length-1; i>=0; i--){
            ans[i] = st.pop();
        }

        return ans;
    }
}